### PR TITLE
fix update project error

### PR DESCRIPTION
**Pull request by QvodSoldier**
#### What type of this PR
/kind bug
#### What this PR does / why we need it:

Fix the problem that if the display name is empty, it will be overwritten by the name when updating the project

#### Which issue(s) this PR fixes:
[update project bug](https://erda-org.erda.cloud/workBench/projects/387/issues/all?id=60277&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMDQzNiJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=436&type=BUG)

#### Specified Reivewers:
/assgin @Effet

### CODE REVIEW

File: modules/cmdb/services/project/project.go [hunk 1]
```diff
@@ -256,43 +253,63 @@ func (p *Project) Update(projectID int64, updateReq *apistructs.ProjectUpdateBod
...
+		return nil, errors.Errorf("failed to update project")
+	}
+
+	return &project, nil
+}
+
+func patchProject(project *model.Project, updateReq *apistructs.ProjectUpdateBody) error {
+	clusterConfig, err := json.Marshal(updateReq.ClusterConfig)
+	if err != nil {
+		logrus.Infof("failed to marshal clusterConfig, (%v)", err)
```
Comment (sfwn) [inline modules/cmdb/services/project/project.go:282]: use correct log level of Errorf.

File: modules/cmdb/services/project/project.go [hunk 2]
```diff
@@ -824,15 +841,18 @@ func checkClusterConfig(clusterConfig map[string]string) error {
...
 }
 func checkRollbackConfig(rollbackConfig *map[string]int) error {
-	if len(*rollbackConfig) != 4 {
-		*rollbackConfig = map[string]int{
-			string(types.DevWorkspace):     5,
-			string(types.TestWorkspace):    5,
-			string(types.StagingWorkspace): 5,
-			string(types.ProdWorkspace):    5,
-		}
+	// DEV/TEST/STAGING/PROD四个环境集群配置
```
Comment (sfwn) [inline modules/cmdb/services/project/project.go:844]: All new comments use english please.

File: modules/cmdb/services/project/project.go [hunk 3]
```diff
@@ -824,15 +841,18 @@ func checkClusterConfig(clusterConfig map[string]string) error {
...
+	l := len(*rollbackConfig)
+	// 空则不配置
```
Comment (sfwn) [inline modules/cmdb/services/project/project.go:846]: same.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/cmdb/services/project/project.go (hunk 1 above)
Comment (QvodSoldier) [reply modules/cmdb/services/project/project.go]: Has been changed

File: modules/cmdb/services/project/project.go (hunk 3 above)
Comment (QvodSoldier) [reply modules/cmdb/services/project/project.go]: Has been changed

**Review by sfwn — APPROVED** [review]

**Pull request merged**